pkg/proxy/proxy.go

```diff
@@ -496,43 +505,67 @@ func (r *RuntimeProxy) containerStats(ctx context.Context, method string, req, r
 	return resp, nil
 }
 
-func (r *RuntimeProxy) handleImage(ctx context.Context, method string, req, resp CRIObject) (interface{}, error) {
+// We don't want to force the user to prefix image names so instead, prefer
+// to say the image is not present if it's not available to all CRIs
+func (r *RuntimeProxy) handleImageStatus(ctx context.Context, method string, req, resp CRIObject) (interface{}, error) {
 	in := req.(ImageObject)
-	client, unprefixed, err := r.clientForImage(in.Image(), true)
-	if client == nil {
-		// the client is offline
-		return resp, nil
+	for i := range r.clients {
+		client, err := r.clientAtIndex(i)
+		if err != nil {
+			continue
+		}
+		imageName := in.Image()
+		_, err = client.invokeWithErrorHandling(ctx, method, req, resp)
+		if err != nil {
+			glog.Errorf("Error in ImageStatus for client %s: %v", client.getID(), err)
+			return nil, err
+		}
+		if out, ok := resp.(ImageStatusResponse); ok && out.Image() != nil {
+			img := out.Image().(Image)
+			if len(img.RepoDigests()) > 0 {
+				imageName = img.RepoDigests()[0]
+			}
+			r.setImageNameById(img.Id(), imageName, true)
+			out.SetImage(out.Image())
+			// If our Id is nil, we don't have the image so just return
+			// immediately so we tell k8s we need to pull the image
+			if out.Image().Id() == "" {
+				glog.Infof("ImageStatus: empty image id in client %s", client.getID())
+				return resp, nil
+			}
+		}
 	}
-	in.SetImage(unprefixed)
+	return resp, nil
```

Comment:
This will return the response from the last CRI, right? But some CRI might expect the digest to be set in the podspec instead of the bare image name. The kubelet updates the image field for containers to be the digest before calling CreateContainer(); however, if resp here is from kiyot (which does not set RepoDigests), then that won't happen.

So I think it's safer to give back the response here that has the digest set (if there's any response with that info).